src/plugins/DevTools/DevTools.ts

```diff
@@ -4,9 +4,25 @@ import devtoolsTemplates from './DevTools.xml';
 import { OwlUIComponent } from '../../ui/OwlUIComponent';
 
 export class DevTools extends JWOwlUIPlugin {
-    constructor() {
-        super();
-        this.componentsRegistry.push(DevToolsComponent as typeof OwlUIComponent);
-        this.templates = devtoolsTemplates;
+    static templates = devtoolsTemplates;
+    Components = [DevToolsComponent as typeof OwlUIComponent];
+    constructor(dispatcher, env) {
+        super(dispatcher, env);
+        this._mountAndRegisterComponents(); // called instead of init
+    }
+
+    //--------------------------------------------------------------------------
+    // Private
+    //--------------------------------------------------------------------------
+
+    /**
+     * Mount and register all components.
+     */
+    _mountAndRegisterComponents(): void {
+        const components = this._instantiateComponents(this.Components);
+        components.forEach((component: OwlUIComponent<{}, {}>): void => {
+            this.env.editor.dispatcher.register(component.handlers, component.commands);
+        });
+        this._mountComponents(components);
```

Comment:
I'm sorry I don't remember: Why can't all JWOwlUIPlugin behave like this ? In the sense of "Why can't this be the default behavior ?"

Reply:
I would like it to be. But for that any extension of `JWOwlUIPlugin` should declare its `Components` _before_ calling `super` (so that `JWOwlUIPlugin` can mount the components). Typescript doesn't allow anything to happen before `super`.
Now the specific case of _registering_ the components in the dispatcher is very specific to `DevTools`: we should not make it the default that UI plugins listen to actions - there is no reason for them to do that. `DevTools` is kind of a weird beast on that level: it's a UI plugin that also behaves a bit like a regular plugin...